What: Initiate the TOC Operations folder (#1234)
cncf/toc: operations/onboarding.md

If, for whatever reason, a TOC member feels they cannot sustain the commitment required of the role, they may proactively notify the TOC Chair or CNCF Staff of their decision to step down.  While uncommon, this is entirely reasonable to occur, particularly as lives and careers grow and shift. We thank each TOC member for their time of service to the community, however long or brief it may be.

### Communicating effectively for consensus

> One of the most critical aspects of the TOC is our ability to facillitate effective communication, particular among ourselves. The TOC holds many discussions over Slack. Members do their best to summarize content and may use emojis  to indicate types of information for quick review or action. An effective tactic that is being normalized is:

Reviewer: ```suggestion
One of the most critical aspects of the TOC is our ability to facillitate effective communication, particular among ourselves. The TOC holds many discussions over Slack. Members do their best to summarize content and may use emojis to indicate types of information for quick review or action. An effective tactic that is being normalized is:
```

### Health and alignment

The TOC may choose to review a project as a result of a health indicator, project event, or other occassion. These reviews are typically specific to the occassion that warranted them but usually involves, researching the problem or indicator, understanding if the metric or occassion is accurate in its depiction and warrants attention, action, or other next step.

When a health or alignment concern is elevated to the TOC, TOC members should adhere to the [project health review guide](project-health-review.md).

## Voting

> The TOC primarily votes in one of two mechanisms, gitvote or on the mailing list. It is our preference that voting occur with [gitvote](https://github.com/cncf/gitvote) in the repository as this provides better transparency and is easier to verify and track vote execution. Voting conducted over the mailing list is performed by responding with a +1 Binding (+1B), or -1 Binding (-1). TOC members may abstain if they have a conflict of interest. Indication of abstention from voting for any other reason is discouraged and should not be confused with not yet voting as indication of abstention is active, not passive, and results in no vote being rendered by that individual. In the absence of information, we cannot make an informed decision, so it is imperative that we ask questions openly and provide answers with clarity for our peer TOC members, particularly on topics that may not be their strength. A vote passes with two-thirds vote of votes cast based on the charter (see [6(c)(viii)](https://www.cncf.io/about/charter)). In the rare event that an issue requiring a vote is senstive, the TOC will leverage the private mailing list to conduct the vote with the resulting decision made public.

Reviewer: Any reason why we need 2 mechanisms? I agree gitvote is easier and I should have send Nic Jackson's tag-network co-chair as gitvote if I knew this earlier.

Author: it developed that way. Prior, we used the mailing list to for conducting voting. A lot of people are on the mailing list and provides an opportunity for subscribers to support those changes and nominations.  Gitvote however makes it much cleaner for the TOC to make decisions on specific items asynchronously, there is less manually counting involved and votes are not lost as we can see whose vote is still outstanding. Its more transparent.

We should always have a redundant system incase the primary doesnt work. Albeit we should consider what types of decisions warrant a mailing list vote versus notification and how we can redirect traffic to the TOC repo where those decisions are reflected.

Reviewer 2: I think we should make sure that we are flexible with voting as long as it's transparent. I want to avoid projects voting on private mailing lists which I see sometimes and we should lead with example.

I do think GitVote makes things super easy and we should use it for more TOC activities

## Lobbying and Direct Messages

> Historically, the TOC has had a problem with “lobbying”, e.g. a project approaching individual TOC members for feedback / comment, often  moving on to another TOC member if they didn’t get the answer they wanted. This was particularly true for the old Sandbox model and was a key reason for changing from sponsorship to a voting model. With this in mind, TOC members should disclose to the rest of the TOC if they have been approached by a project or individual within the community or have any hard or soft conflicts of interest. It is acceptable and expected that projects or community members will reach out to the TOC for advice, however to maintain consistency in expectations, *all TOC members should be made aware of those discussions, with the opportunity to provide additional information or recommendations*. This ensures the TOC maintains a strong, clear voice to our technical communtiy, consistent with our processes, [values](https://github.com/cncf/foundation/blob/main/charter.md#3-values), [principles](../PRINCIPLES.md), and the Charter.

Reviewer: ```suggestion
Historically, the TOC has had a problem with “lobbying”, e.g. a project approaching individual TOC members for feedback / comment, often moving on to another TOC member if they didn’t get the answer they wanted. This was particularly true for the old Sandbox model and was a key reason for changing from sponsorship to a voting model. With this in mind, TOC members should disclose to the rest of the TOC if they have been approached by a project or individual within the community or have any hard or soft conflicts of interest. It is acceptable and expected that projects or community members will reach out to the TOC for advice, however to maintain consistency in expectations, *all TOC members should be made aware of those discussions, with the opportunity to provide additional information or recommendations*. This ensures the TOC maintains a strong, clear voice to our technical communtiy, consistent with our processes, [values](https://github.com/cncf/foundation/blob/main/charter.md#3-values), [principles](../PRINCIPLES.md), and the Charter.
```

Reviewer: fyi - deleted an extra space before `moving`

> As the TOC continues to work through updating our documentation on the repo, older references to "seeking sponsorship" may result in projects reaching out to TOC members directly to "sponsor" their project for moving levels.  When this occurs, kindly direct them to the Moving Levels resources which details how projects are recommended for promotion to the next level.  The TOC strives to pick up projects from the backlog in the order which they are recommended for promotion, while balancing alignment of TOC member expertise, liaisoning assignment, and availability of our members. 

Reviewer: would be good to have a link to "moving levels resources"

## Public persona and engagement

> As a member of the TOC, you may receive requests to speak at conferences, community days or events, or even on podcasts and webcasts and contribute to or participate in panels, blogs, and articles. It is important that you are aware of the capacity that you are participating in these activities in.  Depending on the activity, particularly for conferences and community days or events facillitated by the CNCF, support may be available to allow a TOC member to attend and speak. If you are approached to do this for a CNCF facillitated event and interested, reach out to CNCF Staff.

Reviewer: would be good to specify which CNCF staff or a mailing list?

Reviewer 2: I think CNCF staff is good enough

# Welcome to the TOC!

> We are so glad you are here. You have been elected to the Technical Oversight Committee, the body responsible for accepting and overseeing projects in the cloud native ecosystem.  This onboarding document and its various files and links are here to help get you up to speed and set you up for success as a TOC member. If you haven't already done so, now is a great time to familiarize yourself with the [Definition of Cloud Native](../DEFINITION.md).

Reviewer 3: change to "the body responsible for reviewing, accepting/rejecting, and overseeing projects..."

## Attendence, Availability, and Involvement

The expected time commitment of a TOC member is anywhere from **10-25% of their monthly regular work time** and will vary month to month with surges and slow downs throughout their term. TOC members should also anticipate travel to conferences and offsites at least twice a year, with participation and attendence at various meetings during the CNCF flagship conference: KubeCon+CloudNativeCon (KCCN).

The TOC has four (4) hour-long meetings per month held on Tuesdays at 0800 PST/1100 EST/1700 CET.  The first meeting of each month are Technical Advisory Group (TAG) updates and discussion. As a [Liaison](../../tags/cncf-tags.md#toc-liaisons) to one or more TAGs, the content presented and discussed by the TAG should be information you are already aware of. For TAGs that you are not a liaison for, this is your opportunity to become aware of the broader work being done within speciality technical domains and verticals.  It allows you to connect various TAGs and projects together, promoting interoperability for the benefit of the technical community and its adopters.

> The second and fourth meetings are generally private meetings. During these meetings, the TOC meets to review issues on the repo, conduct vote parties (to ensure outstanding votes are completed), and discuss any open issues, activities, tasks, and other efforts. Periodically, we may invite the Maintainers, projects or other groups to join us to join so we may connect with them, understand any challenges and successes they encounter, and take feedback to improve our engangements.

Reviewer 4: ```suggestion
The second and fourth meetings are generally private meetings. During these meetings, the TOC meets to review issues on the repo, conduct vote parties (to ensure outstanding votes are completed), and discuss any open issues, activities, tasks, and other efforts. Periodically, we may invite the Maintainers, projects, or other groups to join us to join so we may connect with them, understand any challenges and successes they encounter, and take feedback to improve our engangements.
```

The third meeting is public and will usually cover public discussion on improvements and activities the TOC is pursuing or community-led topics where the TOC has oversight or is needed to facilitate neutral consensus.

> TOC members rotate meeting facilitation as often as we can to ensure the TOC Chair is not always doing it, they already have a lot of tasks, see [TOC Chair](toc-chair.md) for more detail. It is important that the Agenda be set in advance but have the opportunity for flexibility should more pressing issues arise. 

Author: thats a good reference. Our meetings are structured a bit different but I agree this is an area we can improve in the onboarding documentation.

### Events and additional meetings

#### Governing Board

> The TOC attends Governing Board meetings periodically when invited, our participation is typically limited to 1 hr ([except for the TOC Chair](toc-chair.md#Governing-board-meetings)). The most common of these is the Strategy Meeting typically held during KubeCon+CloudNativeCon (KCCN). As the overseeing technical body, it is anticipated the TOC guides any technical strategy discussions. The TOC collectively crafts the messaging and structure for this.

Author: its.... not consistent and slightly chaotic 😅 but i'll provided a bit more clarity.

#### KubeCon+CloudNativeCon (KCCN)

At KCCN, the TOC has three additional engagements; TOC Keynote, TOC Panel, and TOC/TAG Meeting.  The TOC keynote is about 15 minutes in length and is delivered by one or more TOC members and may include guests. Common topics are:
* introducing what the TOC is
* conveying changes to the ecosystem/landscapre since the last KCCN
* community oriented topics

> The TOC Panel is a panel of the TOC members present at the conference and functions similar to a ask-me-anything session. The CNCF CTO initiates the panel by providing an overview to attendees of who the TOC is and what we do. They will often initiate the first few questions to the panel and facilitate attendee questions to the TOC. 

Reviewer 4: ```suggestion
The TOC Panel is a panel of the TOC members present at the conference and functions similar to an ask-me-anything session. The CNCF CTO initiates the panel by providing an overview to attendees of who the TOC is and what we do. They will often initiate the first few questions to the panel and facilitate attendee questions to the TOC. 
```

In addition to these meetings, TOC members are expected to meet synchronously or asynchronously with the TAGs for which they liaison, occassionally attending TAG meetings.

### Inactivity of a TOC member

> If, for any reason, a TOC member is continuously absent without notice, see 6(f)iii, their voting privileges are *suspended* until such time as their consecutive attendence at two meetings. Due to the nature of our work, the inability of a TOC member to cast an *informed* vote (as occurs by meeting attendence *and* involvement in regular engagement as detailed above) penalizes the entire TOC's ability to regularly achieve quorum.  If after the 4th absence the TOC is not notified, the TOC Chair with support of CNCF Staff will make an attempt to address the absence with the TOC member.  If the Chair and Staff are unsuccessful, the Chair will bring the issue of absence to TOC vote in accordance with z6(f)ii at the next private meeting and provide notice to the Governing Board of the outcome. 

Reviewer 5: I don't think it has come up but should there be a TOC chair back up or TOC vice chair in case the the chair is not able move things along (The TOC chair may be on vacation or they may have some other priorities)  Or that responsibility remains with the CNCF staff?  (This may be a discussion item)

Author: This is something OpenSSF has and would be a new concept for the CNCF TOC.  Having been absent from the ecosystem due to unforeseen circumstance in that past, this would definitely provide value if it were to exist. I don't know if the charter needs to identify a vice chair or if the TOC can simply say "in their absence".  @amye ?

Reviewer 6: CNCF staff in the past if the chair is out

Reviewer 7: I would like the TOC to explore the idea of a VC (vice chair). Currently, most of the load is on the TOC chair to lead the TOC, interact with the GB, prep for all of these meetings etc. This was working with the initial workload within the ecosystem, but now we have been expanding substantially, hence might be worthwhile to share the load and have someone to shadow and learn these leadership skills

Reviewer 7: Some o[f these responsibilities](https://github.com/cncf/toc/pull/1234/files#diff-93190491f696d97a11813d3c6229d7c4b24c5c06267c496d320d8832cd3b64aaR13) would be great to share with a TOC VC e.g. TAG-TOC interaction

Author: Leaving this comment open as this is a topic of discussion within the TOC.

Reviewer: Agreed TOC VC would be good to help offload some work from TOC chair and step in if TOC chair is not avail.

> If, for whatever reason, a TOC member feels that cannot sustain the committment required of the role, they may proactively notify the TOC Chair or CNCF Staff of their decision to step down.  While uncommon, this is entirely reasonable to have happen, particularly as lives and careers grow and shift. We thank each TOC member for their time of service to the community, however long or brief it may be.

Reviewer 8: ```suggestion
If, for whatever reason, a TOC member feels they cannot sustain the commitment required of the role, they may proactively notify the TOC Chair or CNCF Staff of their decision to step down. While uncommon, this is entirely reasonable to have to happen, particularly as lives and careers grow and shift. We thank each TOC member for their time of service to the community, however long or brief it may be.
```

* `:exclamation:` (red exclamation point) This emoji is used to indicate an action is needed by the TOC, either for an individual or for the group.

Emojis can be used individually or with other emojis, this usually happens to indicate a reminder has an action with it or a reminder is informational but an update may make it more topical. It is common practice to **bold** the topic immediately following the emoji for longer updates, such as summarization of a meeting.

 We operate on lazy consensus, if a TOC member is not participating in a discussion, it is their responsibility to keep themselves informed of the outcome and impetus of these discussion to guide future decision making as part of their participation in the TOC, the use of emojis and summarization of topics is intended to facillitate this in the most effective manner for a group of time-constrained individuals.

> ### Due Diligence

Author: Good call out, i think i had it in another file that i didnt add to this PR. Adding to my todo.

> The primary work a TOC member engages is in conducting the due diligence of cloud native projects as they advance in their maturity, this process is commonly referred to the moving levels or maturation process. After a TOC member steps forward to sponsor a project, it is anticipated that a kick-off meeting is held to set expectations for what the process will be like, provide rough timelines, collect additional information and allow the project to ask any questions.  This kick-off is commonly conducted as a virtual meeting, not lasting more than an hour. Additionally, TOC members should anticipate a few hours of concentrated review, analysing the project, its repository(ies), and clarifying the content of the due diligence. Additional time is expected to be spent conducting and summarizing Adopter interviews to ascertain the current usage of the project within industry.

Reviewer 4: ```suggestion
The primary work a TOC member engages is in conducting the due diligence of cloud native projects as they advance in their maturity, this process is commonly referred to the moving levels or maturation process. After a TOC member steps forward to sponsor a project, it is anticipated that a kick-off meeting is held to set expectations for what the process will be like, provide rough timelines, collect additional information and allow the project to ask any questions.  This kick-off is commonly conducted as a virtual meeting, not lasting more than an hour. Additionally, TOC members should anticipate a few hours of concentrated review, analysing the project, its repository(ies), and clarifying the content of the due diligence. Additional time is expected to be spent conducting and summarizing Adopter interviews to ascertain the current usage of the project within the industry.
```

## Voting

> The TOC primarily votes in one of two mechanisms, gitvote or on the mailing list. It is our preference that voting occur with gitvote in the repository as this provides better transparency and is easier to verify and track vote execution. Voting conducted over the mailing list is performed by responding with a +1 Binding (+1B), or -1 Binding (-1). TOC members may abstain if they have a conflict of interest. Indication of abstention from voting for any other reason is discouraged and should not be confused with not yet voting as indication of abstention is active, not passive, andresults in no vote being rendered by that individual. In the absence of information, we cannot make an informed decision, so it is imperative that we ask questions openly and provide answers with clarity for our peer TOC members, particularly on topics that may not be their strength. A vote passes with two-thirds vote of votes cast based on the charter (see [6(c)(viii)](https://www.cncf.io/about/charter)).

Reviewer 4: ```suggestion
The TOC primarily votes in one of two mechanisms, gitvote or on the mailing list. It is our preference that voting occur with gitvote in the repository as this provides better transparency and is easier to verify and track vote execution. Voting conducted over the mailing list is performed by responding with a +1 Binding (+1B), or -1 Binding (-1). TOC members may abstain if they have a conflict of interest. Indication of abstention from voting for any other reason is discouraged and should not be confused with not yet voting as indication of abstention is active, not passive, and results in no vote being rendered by that individual. In the absence of information, we cannot make an informed decision, so it is imperative that we ask questions openly and provide answers with clarity for our peer TOC members, particularly on topics that may not be their strength. A vote passes with two-thirds vote of votes cast based on the charter (see [6(c)(viii)](https://www.cncf.io/about/charter)).
```

> The TOC primarily votes in one of two mechanisms, gitvote or on the mailing list. It is our preference that voting occur with gitvote in the repository as this provides better transparency and is easier to verify and track vote execution. Voting conducted over the mailing list is performed by responding with a +1 Binding (+1B), or -1 Binding (-1). TOC members may abstain if they have a conflict of interest. Indication of abstention from voting for any other reason is discouraged and should not be confused with not yet voting as indication of abstention is active, not passive, and results in no vote being rendered by that individual. In the absence of information, we cannot make an informed decision, so it is imperative that we ask questions openly and provide answers with clarity for our peer TOC members, particularly on topics that may not be their strength. A vote passes with two-thirds vote of votes cast based on the charter (see [6(c)(viii)](https://www.cncf.io/about/charter)).

Reviewer 7: I do not recall having a mailing list vote after introducing the gitvote. Will we have any issues that require voting that cannot be solved in a GH issue with gitvote?

Author: I dont believe so as our decision making that requires votes should be documented/recorded and public. Thinking through sandbox applications, we used to vote on the call which was recorded and @amye tallied/tracked votes. Since we've moved many of our processes to more transparent mediums (issues, PRs, etc.) that allows for gitvote, I do not believe we have any left that require facilitation of a vote outside of GitHub.

Reviewer 7: Perhaps we can move it as a secondary vote option, perhaps for sensitive issues requiring a closed vote (but cannot think of any scenarios where that was required)- aka a backup voting solution.

Reviewer 8: We could add a link to https://github.com/cncf/gitvote

## Lobbying and Direct Messages

> Historically, the TOC has had a problem with “lobbying”, e.g. a project approaching individual TOC members for feedback / comment, often  moving on to another TOC member if they didn’t get the answer they wanted. This was particularly true for the old Sandbox model and was a key reason for changing from sponsorship to a voting model. With this in mind, TOC members should disclose to the rest of the TOC if they have been approached by a project or individual within the community. It is acceptable and expected that projects or community members will reach out to the TOC for advice, however to maintain consistency in expectations, *all TOC members should be made aware of those discussions, with the opportunity to provide additional information or recommendations*. This ensures the TOC maintains a strong, clear voice to our technical communtiy, consistent with our processes, [values](https://github.com/cncf/foundation/blob/main/charter.md#3-values), [principles](../PRINCIPLES.md), and the Charter.

Reviewer 9: @TheFoxAtWork Do we want to add anything here about TOC members disclosing and/or recusing themselves when they have an extra stake (financial? investment?) in the outcome of a TOC vote/decision?

Author: let me add something in to that effect.
